src/pages/home/report/ReportActionsView.js

```diff
@@ -497,6 +503,27 @@ class ReportActionsView extends React.Component {
     }
 
     render() {
+        const isDefaultChatRoom = isDefaultRoom(this.props.report);
+        const participants = lodashGet(this.props.report, 'participants', []);
+        const isMultipleParticipant = participants.length > 1;
+        const displayNamesWithTooltips = _.map(
+            getPersonalDetailsForLogins(participants, this.props.personalDetails),
+            ({
+                displayName, firstName, login, pronouns,
+            }) => {
+                const displayNameTrimmed = Str.isSMSLogin(login) ? this.props.toLocalPhone(displayName) : displayName;
+
+                return {
+                    displayName: (isMultipleParticipant ? firstName : displayNameTrimmed) || Str.removeSMSDomain(login),
+                    tooltip: Str.removeSMSDomain(login),
+                    pronouns,
+                };
+            },
+        );
+
+        const chatUsers = isDefaultChatRoom ? [{displayName: this.props.report.reportName}] : displayNamesWithTooltips;
```

Comment:
I feel that this code block can be moved into `ChatBeginingText`. and you can just pass the `report` to the `ChatBeginingText`. 

Thus we would be able to remove the `personalDetails` subscription from here.

Reply:
ok got it. Moved this part code to ChatBeginingText.